Sources/NIOPosix/DatagramVectorReadManager.swift

```diff
@@ -178,7 +178,9 @@ struct DatagramVectorReadManager {
 #else
             precondition(self.messageVector[i].msg_hdr.msg_namelen != 0, "Unexpected zero length peer name")
 #endif
-            let address: SocketAddress = self.sockaddrVector[i].convert()
+            let address: SocketAddress
+            do { address = try self.sockaddrVector[i].convert() }
+            catch { fatalError("Socket address conversion failed.") }
```

Comment:
More broadly, should we tolerate this error in this location?

Reply:
That's a great question.

I suppose we have to choose between failing hard or silently dropping these messages, given that we will be unable to construct an `AddressedEnvelope` without a `SocketAddress`.

I opted to fail because:
1. It's closest to the behaviour before this patch—where it would have still crashed, just inside `convert()`.
2. Above these lines there are a number of `precondition` checks so failing hard seemed to be "the done thing" here.

Happy to revisit though if you think this is the wrong choice here.

Reply:
I think there's a third option, which is to throw. The preconditions above are correctness checks, but this is not. Throwing here is consistent with the scalar read path.

Reply:
****